src/cpu/aarch64/cpu_reducer.cpp

```diff
@@ -217,7 +217,11 @@ struct reducer_2d_driver_f_s_32_t : public reducer_2d_driver_t<data_type, isa> {
         const int load_len[nbranches] = {vlen, vlen, typesize};
         Label loop_x_label[nbranches + 1];
 
-        this->ptrue(preg_all.b);
+        switch (isa) {
```

Comment:
I don't have a good understanding of xbyak for SVE, sorry, but do you know why this is necessary? Why can't we just use plain old ptrue?

Reply:
Hi @jondea, we edited this code to support multiple ISA vector lengths for our kernels.

Reply:
I understand that was the general idea, but why is `VL32` and `VL64` needed, rather than the default `ALL`? Was it so that you could test the 256 kernels on a machine with 512 vector length?

Reply:
yes, you got it right it right. It enables sve_256 kernels on 512 VL machine.